NoSuchMethodError with 203.4449 EAPs

<!--
Hello and thank you for the issue!
If you would like to report a bug, we have added some points below that you can fill out.
Feel free to remove all the irrelevant text to request a new feature.
-->

## Environment

* **IntelliJ Rust plugin version:** 0.3.133.3402-203-nightly
* **Rust toolchain version:** 1.47.0 (18bf6b4f0 2020-10-07) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.3 EAP Ultimate Edition (IU-203.4449.2)
* **Operating system:** macOS 10.15.6

## Problem description

<details>

```
Error during macro expansion

java.lang.NoSuchMethodError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:603)
	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at kotlin.streams.jdk8.StreamsKt.toList(Streams.kt:60)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1.invoke(MacroExpansionTask.kt:202)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1.invoke(MacroExpansionTask.kt:48)
	at org.rust.stdext.CompletableFutureKt$supplyAsync$1.get(CompletableFuture.kt:18)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoSuchMethodError: 'void com.intellij.util.containers.ConcurrentWeakKeySoftValueHashMap.<init>(int, float, int, gnu.trove.TObjectHashingStrategy)'
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt$createWeakMap$1.<init>(RsResolveCache.kt:270)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.createWeakMap(RsResolveCache.kt:274)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.getOrCreateMap(RsResolveCache.kt:264)
	at org.rust.lang.core.resolve.ref.RsResolveCacheKt.access$getOrCreateMap(RsResolveCache.kt:1)
	at org.rust.lang.core.resolve.ref.RsResolveCache.getRustStructureDependentCache(RsResolveCache.kt:57)
	at org.rust.lang.core.resolve.ref.RsResolveCache.getCacheFor(RsResolveCache.kt:151)
	at org.rust.lang.core.resolve.ref.RsResolveCache.resolveWithCaching(RsResolveCache.kt:97)
	at org.rust.lang.core.resolve.ref.RsReferenceCached.cachedMultiResolve(RsReferenceCached.kt:27)
	at org.rust.lang.core.resolve.ref.RsReferenceCached.multiResolve(RsReferenceCached.kt:20)
	at com.intellij.psi.PsiPolyVariantReferenceBase.resolve(PsiPolyVariantReferenceBase.java:47)
	at org.rust.lang.core.resolve.ref.RsReferenceBase.resolve(RsReferenceBase.kt:28)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion$getDeclarationsFor$1$1.process(RsModulesIndex.kt:46)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion$getDeclarationsFor$1$1.process(RsModulesIndex.kt:31)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:73)
	at com.intellij.psi.stubs.StubIndexImpl.lambda$processElements$2(StubIndexImpl.java:289)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:325)
	at com.intellij.psi.stubs.StubIndex.processElements(StubIndex.java:49)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion$getDeclarationsFor$1.invoke(RsModulesIndex.kt:43)
	at org.rust.lang.core.stubs.index.RsModulesIndex$Companion.getDeclarationsFor(RsModulesIndex.kt:52)
	at org.rust.lang.core.psi.RsFile$declarations$1.compute(RsFile.kt:206)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.psi.RsFile.getDeclarations(RsFile.kt:205)
	at org.rust.lang.core.psi.RsFile.getDeclaration(RsFile.kt:190)
	at org.rust.lang.core.psi.RsFile.doGetCachedData(RsFile.kt:101)
	at org.rust.lang.core.psi.RsFile.access$doGetCachedData(RsFile.kt:68)
	at org.rust.lang.core.psi.RsFile$cachedData$1$value$1.compute(RsFile.kt:93)
	at org.rust.lang.core.psi.RsFile$cachedData$1$value$1.compute(RsFile.kt:68)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at org.rust.openapiext.UtilsKt.recursionGuard(utils.kt:69)
	at org.rust.openapiext.UtilsKt.recursionGuard$default(utils.kt:68)
	at org.rust.lang.core.psi.RsFile$cachedData$1.compute(RsFile.kt:93)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:153)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:54)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$1(CachedValueBase.java:235)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:236)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:43)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:76)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:150)
	at org.rust.lang.core.psi.RsFile.getCachedData(RsFile.kt:92)
	at org.rust.lang.core.psi.RsFile.getCrateRoot(RsFile.kt:77)
	at org.rust.lang.core.macros.SourceFile.extract(ExpandedMacroStorage.kt:535)
	at org.rust.lang.core.macros.SourceFile.extract(ExpandedMacroStorage.kt:500)
	at org.rust.lang.core.macros.Extractable.extract(MacroExpansionTask.kt:329)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1$1.invoke(MacroExpansionTask.kt:191)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1$1.invoke(MacroExpansionTask.kt:48)
	at org.rust.openapiext.UtilsKt$runReadActionInSmartMode$1.compute(utils.kt:275)
	at com.intellij.openapi.project.DumbService.lambda$runReadActionInSmartMode$0(DumbService.java:103)
	at com.intellij.openapi.project.DumbService.lambda$runReadActionInSmartMode$1(DumbService.java:147)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:890)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:140)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:103)
	at org.rust.openapiext.UtilsKt.runReadActionInSmartMode(utils.kt:273)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1.invoke(MacroExpansionTask.kt:188)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1$result$1.invoke(MacroExpansionTask.kt:48)
	at org.rust.openapiext.UtilsKt$executeUnderProgressWithWriteActionPriorityWithRetries$success$1.invoke(utils.kt:285)
	at org.rust.openapiext.UtilsKt$executeUnderProgressWithWriteActionPriorityWithRetries$success$1.invoke(utils.kt)
	at org.rust.openapiext.UtilsKt$sam$java_lang_Runnable$0.run(utils.kt)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:152)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:113)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:59)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:110)
	at org.rust.openapiext.UtilsKt.runWithWriteActionPriority(utils.kt:297)
	at org.rust.openapiext.UtilsKt.executeUnderProgressWithWriteActionPriorityWithRetries(utils.kt:284)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1.apply(MacroExpansionTask.kt:186)
	at org.rust.lang.core.macros.MacroExpansionTaskBase$submitExpansionTask$1$stages1$1.apply(MacroExpansionTask.kt:48)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)
	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)
	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	... 5 more
```

</details>

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -29,7 +29,8 @@ val ideaVersion = prop("ideaVersion")
 val clionVersion = prop("clionVersion")
 val baseVersion = when (baseIDE) {
     "idea" -> ideaVersion
-    "clion" -> clionVersion
+    // FIXME: 203.4203. use clionVersion
+    "clion" -> if (platformVersion == 203) ideaVersion else clionVersion
     else -> error("Unexpected IDE name: `$baseIDE`")
 }
 
@@ -183,7 +184,8 @@ project(":plugin") {
             graziePlugin,
             psiViewerPlugin
         )
-        if (baseIDE == "idea") {
+        // FIXME: 203.4203. use `baseIDE == "idea"`
+        if (baseVersion == ideaVersion) {
             plugins += listOf(
                 "copyright",
                 "java",

--- a/gradle-202.properties
+++ b/gradle-202.properties
@@ -1,11 +1,11 @@
 # Existent IDE versions can be found in the following repos:
 # https://www.jetbrains.com/intellij-repository/releases/
 # https://www.jetbrains.com/intellij-repository/snapshots/
-ideaVersion=IU-2020.2.2
-clionVersion=CL-2020.2.2
+ideaVersion=IU-2020.2.3
+clionVersion=CL-2020.2.4
 
 # https://plugins.jetbrains.com/plugin/12775-native-debugging-support/versions
-nativeDebugPluginVersion=202.7319.45
+nativeDebugPluginVersion=202.7660.3
 # https://plugins.jetbrains.com/plugin/12175-grazie/versions
 graziePluginVersion=202.6948.45
 # https://plugins.jetbrains.com/plugin/227-psiviewer/versions

--- a/gradle-203.properties
+++ b/gradle-203.properties
@@ -1,16 +1,17 @@
 # Existent IDE versions can be found in the following repos:
 # https://www.jetbrains.com/intellij-repository/releases/
 # https://www.jetbrains.com/intellij-repository/snapshots/
-ideaVersion=IU-203.4203-EAP-CANDIDATE-SNAPSHOT
+ideaVersion=IU-203.4449-EAP-CANDIDATE-SNAPSHOT
+# TODO: fix all "FIXME: 203.4203." comments in `build.gradle.kts`
 clionVersion=CL-203.4203-EAP-CANDIDATE-SNAPSHOT
 
 # https://plugins.jetbrains.com/plugin/12775-native-debugging-support/versions
-nativeDebugPluginVersion=203.4203.26
+nativeDebugPluginVersion=203.4449.2
 # https://plugins.jetbrains.com/plugin/12175-grazie/versions
-graziePluginVersion=203.4203.24
+graziePluginVersion=203.4449.8
 # https://plugins.jetbrains.com/plugin/227-psiviewer/versions
 psiViewerPluginVersion=203-SNAPSHOT
 
 # please see http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html for description
-sinceBuild=203.3645
+sinceBuild=203.4449
 untilBuild=211.*

--- a/src/202/main/kotlin/org/rust/lang/core/resolve/ref/compatUtils.kt
+++ b/src/202/main/kotlin/org/rust/lang/core/resolve/ref/compatUtils.kt
@@ -1,0 +1,11 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.util.containers.ContainerUtil
+import gnu.trove.TObjectHashingStrategy
+
+fun <T> canonicalStrategy(): TObjectHashingStrategy<T> = ContainerUtil.canonicalStrategy()

--- a/src/203/main/kotlin/org/rust/lang/core/resolve/ref/compatUtils.kt
+++ b/src/203/main/kotlin/org/rust/lang/core/resolve/ref/compatUtils.kt
@@ -1,0 +1,11 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.util.containers.HashingStrategy
+
+@Suppress("UnstableApiUsage")
+fun <T> canonicalStrategy(): HashingStrategy<T> = HashingStrategy.canonical()

--- a/src/main/kotlin/org/rust/lang/core/resolve/ref/RsResolveCache.kt
+++ b/src/main/kotlin/org/rust/lang/core/resolve/ref/RsResolveCache.kt
@@ -26,7 +26,6 @@ import com.intellij.psi.util.CachedValueProvider
 import com.intellij.psi.util.CachedValuesManager
 import com.intellij.psi.util.PsiUtilCore
 import com.intellij.util.containers.ConcurrentWeakKeySoftValueHashMap
-import com.intellij.util.containers.ContainerUtil
 import org.rust.lang.core.macros.MacroExpansionManager
 import org.rust.lang.core.psi.*
 import org.rust.lang.core.psi.ext.RsModificationTrackerOwner
@@ -271,7 +270,7 @@ private fun <K, V> createWeakMap(): ConcurrentMap<K, V> {
         100,
         0.75f,
         Runtime.getRuntime().availableProcessors(),
-        ContainerUtil.canonicalStrategy()
+        canonicalStrategy()
     ) {
         override fun createValueReference(
             value: V,

